### PR TITLE
Deprecate Fit.minuit member

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request deprecates the `Fit.minuit` member and moves it to the `OptimizeResult` instead. The motivation is to get rid of the stateful dependency of the `Fit` object, which is likely the reason for the remaining test fails in #4406.

### CODE REVIEW

File: gammapy/modeling/fit.py [hunk 1]
```diff
@@ -249,6 +256,9 @@ def covariance(self, datasets):
         ----------
         datasets : `Datasets` or list of `Dataset`
             Datasets to optimize.
+        optimize_result : `OptimizeResult`
+            Optimization result. Can be optionally used to pass the state of the IMinuit object
+            to the convariance estimatiomn. This might save computation time in certain cases.
```
Comment (registerrier) [inline gammapy/modeling/fit.py:261]:
```suggestion
            to the covariance estimation. This might save computation time in certain cases.
```

**Review by registerrier — DISMISSED** [review]
Thanks @adonath . Looks good. No comment from my side except for docstring typos.

**Review by registerrier — COMMENTED** [review]
Looking at it again, the deprecation scheme is not really respected since the `_iminuit` attribute is no longer set. 
Is it possible to keep but not use it in tests? Will this prevent the multiprocessing to run properly?

**Comment by adonath** [comment]
@registerrier Good point! No, I think I can just add it back...

**Comment by adonath** [comment]
@registerrier I added test to access the private member `._minuit`.

**Review by QRemy — APPROVED** [review]
Thanks @adonath, no comments from my side.

**Pull request merged**